### PR TITLE
Enable schema validation for sdp

### PR DESCRIPTION
**Pull request by vicancy**
Enable schema validation for sdp

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    using Newtonsoft.Json.Linq;
+    using Newtonsoft.Json.Schema;
+
+    public class SchemaValidateService
+    {
+        private static ConcurrentDictionary<string, SchemaValidateService> _cache = new ConcurrentDictionary<string, SchemaValidateService>();
+
+        private bool _schemaValidationEnabled = true;
+        private object locker = new object();
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs]: >locker [](start = 23, length = 6)

_locker?

File: src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            {
+                ValidateObject(obj, schema, (sender, args) => errors.Add(args.Message));
+            }
+            catch (Exception e)
+            {
+                if (_schemaValidationEnabled)
+                {
+                    lock (locker)
+                    {
+                        if (_schemaValidationEnabled)
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs]: >_schemaValidationEnabled [](start = 28, length = 24)

double checked?

Comment (vicancy) [reply src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs]: https://en.wikipedia.org/wiki/Double-checked_locking

File: src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            {
+                return;
+            }
+
+            var errors = new List<string>();
+            try
+            {
+                ValidateObject(obj, schema, (sender, args) => errors.Add(args.Message));
+            }
+            catch (Exception e)
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs]: >Exception [](start = 19, length = 9)

Detail exception?

File: src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.Exceptions;
+
+    using Newtonsoft.Json.Linq;
+    using Newtonsoft.Json.Schema;
+
+    public class SchemaValidateService
+    {
+        private static ConcurrentDictionary<string, SchemaValidateService> _cache = new ConcurrentDictionary<string, SchemaValidateService>();
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs]: readonly

File: src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+        private bool _schemaValidationEnabled = true;
+        private object _locker = new object();
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs]: readonly

File: src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs (hunk 2 above)
Comment (vicancy) [reply src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs]: [https://en.wikipedia.org/wiki/Double-checked_locking](https://en.wikipedia.org/wiki/Double-checked_locking)

---
In reply to: [152904504](https://github.com/dotnet/docfx/pull/2228#discussion_r152904504) [](ancestors = 152904504)

File: src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        private readonly object _locker = new object();
+        private bool _schemaValidationEnabled = true;
+
+        public static readonly SchemaValidateService Instance = new SchemaValidateService();
+
+        private SchemaValidateService()
+        {
+        }
+
+        public void RegisterLicense(string license)
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs]: static?

File: src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+    using Newtonsoft.Json.Linq;
+    using Newtonsoft.Json.Schema;
+
+    public class SchemaValidateService
+    {
+        private static readonly object _sync = new object();
+        private static SchemaValidateService _instance = new SchemaValidateService();
+
+        private readonly object _locker = new object();
+        private bool _schemaValidationEnabled = true;
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/Validators/SchemaValidateService.cs]: `volitale`

**Pull request merged**